x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/JobStorageDeletionTask.java

```diff
@@ -106,6 +113,72 @@ public void delete(String jobId, Client client, ClusterState state,
                 },
                 failureHandler);
 
+        // Step 4(c). If we have any hits, that means we are NOT the only job on this index, and should not delete it
+        ActionListener<SearchResponse> customIndexSearchHandler = ActionListener.wrap(
+            searchResponse -> {
+                if (searchResponse.getHits().totalHits > 0) {
+                    deleteByQueryExecutor.onResponse(true);
+                } else {
+                    logger.info("Running DELETE Index on [" + indexName + "] for job [" + jobId + "]");
+                    DeleteIndexRequest request = new DeleteIndexRequest(indexName);
+                    request.indicesOptions(IndicesOptions.lenientExpandOpen());
+                    executeAsyncWithOrigin(
+                        client.threadPool().getThreadContext(),
+                        ML_ORIGIN,
+                        request,
+                        ActionListener.<AcknowledgedResponse>wrap(
+                            deleteIndexResponse -> deleteAliases(jobId, client, deleteAliasHandler), //skip DBQ
```

Comment:
This is also skipping deleting the aliases. That's OK at present, as each job only uses one results index and if it's deleted then the aliases attached to it will also go. But it would be good to add a comment here saying "also skip deleting aliases as deleting the index will have already deleted them".